### PR TITLE
feat: add support for filtering on list fields

### PR DESCRIPTION
**Pull request by kldavis4**
See #3897

### CODE REVIEW

File: packages/@tinacms/graphql/src/database/datalayer.ts [hunk 1]
```diff
@@ -132,14 +136,26 @@ const makeKeyForField = <T extends object>(
...
       // TODO I think these dates are ISO 8601 so I don't think we need to convert to numbers
       const rawValue = data[field.name]
-      const resolvedValue = String(
-        field.type === 'datetime'
-          ? new Date(rawValue).getTime()
-          : field.type === 'string'
-          ? stringEscaper(rawValue as string | string[])
-          : rawValue
-      ).substring(0, maxStringLength)
-      valueParts.push(applyPadding(resolvedValue, field.pad))
```
Comment (kldavis4) [inline packages/@tinacms/graphql/src/database/datalayer.ts:142]: Note that this change is basically just making explicit what was already happening when an array of strings was cast to a string (ie ['foo','bar'] -> 'foo,bar')

File: packages/@tinacms/graphql/src/database/datalayer.ts [hunk 2]
```diff
@@ -214,6 +230,105 @@ export const coerceFilterChainOperands = (
   return result
 }
```
Comment (kldavis4) [inline packages/@tinacms/graphql/src/database/datalayer.ts:232]: Extracting these functions out so they can be re-used for both the list true/false cases

**Comment by logan-anderson** [comment]
The only thing that I was a bit confused on was the equal value being a string and not an array. 

I would expect `tags: { eq: ['foo', 'bar'] }` but we have `tags: {eq: "foo"}`. Not sure what the correct behaviour is but it did catch me off guard at first.

**Comment by logan-anderson** [comment]
Side note: 

While testing I noticed this bug. Not sure if its related to this PR or not. We can probably don't need to generate filter args for type: 'rich-text'.

Screenshot 2023-05-03 at 2 00 24 PM

**Comment by kingRayhan** [comment]
Is there any update on this PR?

**Comment by kldavis4** [comment]
> Side note:
> 
> While testing I noticed this bug. Not sure if its related to this PR or not. We can probably don't need to generate filter args for type: 'rich-text'.
> 
> Screenshot 2023-05-03 at 2 00 24 PM

not related to this pr. currently it looks like string, reference, number, datetime and boolean are filterable.

**Comment by logan-anderson** [comment]
> Can you elaborate on what you mean? If you had an array of values that you wanted to match against a list value, you'd probably use in and not eq.

What If I wanted to search all of the lists that had exactly ["foo, "bar"].

This might be a bit nuanced and I am totally fine leaving as is.

**Comment by kldavis4** [comment]
> > Can you elaborate on what you mean? If you had an array of values that you wanted to match against a list value, you'd probably use in and not eq.
> 
> What If I wanted to search all of the lists that had exactly ["foo, "bar"].
> 
> This might be a bit nuanced and I am totally fine leaving as is.

I don't think that would be supported and not something I'd recommend supporting unless someone really had a good use case

**Review by logan-anderson — APPROVED** [review]

**Pull request merged**